perceval/components/abstract_processor.py

```diff
@@ -52,35 +47,45 @@ class ProcessorType(Enum):
 
 
 class AProcessor(ABC):
-    def __init__(self):
-        self._input_state = None
-        self.name: str = ""
+    def __init__(self, experiment = None):
+        self.experiment = experiment or Experiment()
+        # TODO: remove the next line when we send experiments on the cloud
         self._parameters: dict[str, any] = {}
+        self._set_min_photons_parameter()
 
-        self._noise: NoiseModel | None = None
+    @property
+    def experiment(self) -> Experiment:
+        return self._experiment
 
-        self._thresholded_output: bool = False  # Deprecated, avoid using this field
-        self._min_detected_photons_filter: int | None = None
+    @experiment.setter
+    def experiment(self, experiment: Experiment):
+        self._experiment = experiment
+        experiment.add_observers(self._circuit_change_observer,
+                                 self._noise_changed_observer,
+                                 self._input_changed_observer)
 
-        self._reset_circuit()
+    @abstractmethod
+    def _circuit_change_observer(self, new_component = None):
+        pass
 
-    def _reset_circuit(self):
-        self._in_ports: dict = {}
-        self._out_ports: dict = {}
-        self._postselect: PostSelect | None = None
+    @abstractmethod
+    def _noise_changed_observer(self):
+        pass
 
-        self._is_unitary: bool = True
-        self._has_td: bool = False
-        self._has_feedforward = False
+    def _input_changed_observer(self):
+        pass
 
-        self._n_heralds: int = 0
-        self._anon_herald_num: int = 0  # This is not a herald count!
-        self._components: list[tuple[tuple, AComponent]] = []  # Any type of components, not only unitary ones
-        self._detectors: list[IDetector] = []
-        self.detectors_injected: list[int] = []  # List of modes where detectors are already in the circuit
-        self._mode_type: list[ModeType] = []
+    @property
+    def name(self) -> str:
+        return self.experiment.name
 
-        self._n_moi: int = 0  # Number of modes of interest (moi)
+    @name.setter
+    def name(self, name: str):
+        self.experiment.name = name
+
+    @property
+    def detectors_injected(self):
```

Comment:
I'd say we can clean up the `AbstractProcessor` even from some public calls (as long as they moved to `Experiment`)
This one can be remove.